python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -1202,6 +1202,51 @@ def test_packing(self):
         assert to_launch == {}
         assert not rem
 
+    def test_skip_placed_bundles(self):
+        # test that we do not launch new nodes for bundles that are already placed.
+        provider = MockProvider()
+        scheduler = ResourceDemandScheduler(
+            provider,
+            TYPES_A,
+            10,
+            head_node_type="p2.8xlarge",
+            upscaling_speed=1,
+        )
+
+        provider.create_node({}, {TAG_RAY_USER_NODE_TYPE: "p2.8xlarge"}, 1)
+        # At this point our cluster has 1 p2.8xlarge instances (8 GPUs) and is
+        # fully idle.
+        nodes = provider.non_terminated_nodes({})
+
+        pending_placement_groups = [
+            PlacementGroupTableData(
+                state=PlacementGroupTableData.PENDING,
+                strategy=PlacementStrategy.PACK,
+                bundles=[
```

Comment:
I am also not sure whether this is possible to happen in runtime.

Reply:
Imagine that the placement group was originally spread across 2 nodes (that is possible with the PACK strategy) but later the second node disappeared. Now, we have 1 node left alive, and if it has enough resources available for the bundle that was originally on the disappeared node this time, then we should not launch a new node.